bin/update.sh

```diff
@@ -49,7 +49,26 @@ if ! echo "$NEW_VERSION" | grep --quiet --extended-regexp '^[0-9]+\.[0-9]+\.[0-9
     exit 1
 fi
 
+UPDATE_IOS=n
+if [ -n "$2" ] && [ "$2" == "--ios-version" ]; then
+    if [ -z "$3" ]; then
+        echo "error: missing argument for --ios-version"
+        exit 1
+    fi
+
+    IOS_VERSION="$3"
+    if ! echo "$IOS_VERSION" | grep --quiet '^v[0-9][0-9]$'; then
+        echo "error: Specified iOS version '${IOS_VERSION}' doesn't match a valid major version (example: \"v15\")"
+        echo "error: Use MAJOR versions only"
+        echo "error: See https://developer.apple.com/documentation/packagedescription/supportedplatform/iosversion for valid iOS versions"
+        exit 1
+    fi
+
+    UPDATE_IOS=y
+fi
```

Comment:
No need for `UPDATE_IOS` separate from `IOS_VERSION`.
Also we shouldn't need to check if `$2` is non-empty.


```suggestion
IOS_VERSION=
if [ "$2" == "--ios-version" ]; then
    if [ -z "$3" ]; then
        echo "error: missing argument for --ios-version"
        exit 1
    fi

    IOS_VERSION="$3"
    if ! echo "$IOS_VERSION" | grep --quiet '^v[0-9][0-9]$'; then
        echo "error: Specified iOS version '${IOS_VERSION}' doesn't match a valid major version (example: \"v15\")"
        echo "error: Use MAJOR versions only"
        echo "error: See https://developer.apple.com/documentation/packagedescription/supportedplatform/iosversion for valid iOS versions"
        exit 1
    fi
fi
```